addon/services/user-service.js

```diff
@@ -94,6 +94,17 @@ export default Service.extend(serviceMixin, {
     return declineInvitation(args);
   },
 
+  getSecuritySettings (portalOpts) {
+    let {username} = this.get('session.currentUser');
+    const urlPath = `/community/users/${username}/forgotPassword?f=json`;
+    return this.request(urlPath, null, portalOpts);
+  },
+
+  generateToken (user, portalOpts, formData) {
+    const urlPath = `/generateToken?f=json`;
```

Comment:
Should this be using https://esri.github.io/arcgis-rest-js/api/auth/generateToken/ ?